tests/core/providers/test_auto_provider.py

```diff
@@ -0,0 +1,40 @@
+from contextlib import contextmanager
+import os
+import pytest
+
+from web3.providers.auto import (
+    load_provider_from_environment,
+)
+from web3.providers import (
+    IPCProvider,
+    HTTPProvider,
+)
+
+
+@contextmanager
+def env(name, tempval):
```

Comment:
You're looking for the pytest `monkeypatch` fixture which does this natively:

https://docs.pytest.org/en/latest/monkeypatch.html